### PR TITLE
Cleanup deprecated analysis.recommendation and unused code

### PR DESCRIPTION
**Pull request by razvanpapadopol**
* Cleanup recommendation from `analysisPreviousSchema` and `analysisNextSchema`
* Remove `RecommendationReason` type
* Remove `recommendationSchema`
* Remove unused `RecommendationWarningToHuman` map
* Remove `RecommendationWarning` type
* Update tests and fixtures

## How has this been tested?

- Automated tests that cover added/changed functionality
- Manual testing with production data (locally)
- Manual testing with mock data (locally or on Vercel)

### CODE REVIEW

**Comment by jessie-ross** [comment]
Thanks for this!

File: src/lib/schemas.ts [hunk 1]
```diff
@@ -356,31 +356,12 @@ export enum AssignmentCacheStatus {
...
 
-export enum RecommendationReason {
-  CiInRope = 'ci_in_rope',
-  CiGreaterThanRope = 'ci_greater_than_rope',
-  CiLessThanRope = 'ci_less_than_rope',
-  CiRopePartlyOverlap = 'ci_rope_partly_overlap',
-  RopeInCi = 'rope_in_ci',
-}
-
 export enum RecommendationWarning {
```
Comment (jessie-ross) [inline src/lib/schemas.ts:359]: This should be deleted too.

**Review by jessie-ross — APPROVED** [review]
Thanks for cleaning this up 🔪

File: src/lib/schemas.ts (hunk 1 above)
Comment (razvanpapadopol) [reply src/lib/schemas.ts:359]: 👍🏼 This means that we can delete also `RecommendationWarningToHuman`, right?
https://github.com/Automattic/abacus/blob/5be596474ebe3da1c99034ddb9070fa718114204/src/lib/analyses.ts#L17-L24

**Pull request merged**